lib/transaction/pool.go

```diff
@@ -93,6 +93,30 @@ func (tp *Pool) Remove(hashes ...string) {
 	}
 }
 
+func (tp *Pool) RemoveFromSources(sources ...string) {
+	if len(sources) < 1 {
+		return
+	}
+
+	tp.Lock()
+	defer tp.Unlock()
+
+	for _, source := range sources {
+		if hash, found := tp.sources[source]; found {
```

Comment:
@Charleslee522 We can remove this nested if ;)
```go
var found bool
var hash string
if hash, found := tp.sources[source]; !found {
	continue
} 

var tx transaction.Transaction
if tx, found = tp.Pool[hash]; !found {
	continue
}

delete(tp.sources, tx.Source())
delete(tp.Pool, hash)

for i, h := range tp.hashes {
	if h != hash {
		continue
	}

	tp.hashes = append(tp.hashes[:i], tp.hashes[i+1:]...)
	break
}
```

Reply:
> We can remove this nested if ;)

@spikeekips Oh, I also don't like to make nested if
but in this case, I figure using `nested if` is simpler.
The above is quite longer than before.

Reply:
@Charleslee522 Yes, it is matter of chice :)